### PR TITLE
nim: fix head build

### PR DESCRIPTION
**Pull request by ghost**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

---

Disabled building the tools for the head build - causes error otherwise.

### CODE REVIEW

File: Formula/nim.rb [hunk 1]
```diff
@@ -21,19 +21,23 @@ def install
...
-    target.install "bin/nimgrep"
-    target.install "bin/nimsuggest"
-
     bin.install_symlink prefix/"nim/bin/nim"
     bin.install_symlink prefix/"nim/bin/nim" => "nimrod"
-    bin.install_symlink prefix/"nim/bin/nimble"
-    bin.install_symlink prefix/"nim/bin/nimgrep"
-    bin.install_symlink prefix/"nim/bin/nimsuggest"
+
+    # Building tools only for non-head branch
```
Comment (MikeMcQuaid) [inline Formula/nim.rb]: Have they been removed? If not, please work with upstream to fix this error with them rather than disabling them.

**Comment by ysalmi** [comment]
The tools are actually part of two related repos that are included in the
point releases but are not present on the head branch. This PR is just a
temporary fix (so that the HEAD build works). I will work with upstream to
find a proper fix.

**Comment by MikeMcQuaid** [comment]
> I will work with upstream to find a proper fix.

Let's wait until you've done that as it may inform the best fix here.

**Comment by ysalmi** [comment]
I worked with the upstream devs to find a good solution. Now HEAD builds just like a normal release.

File: Formula/nim.rb [hunk 2]
```diff
@@ -16,24 +16,28 @@ class Nim < Formula
   def install
     if build.head?
       system "/bin/sh", "bootstrap.sh"
+
+      # Grab the tools source (only included in point releases)
+      system "git", "clone", "https://github.com/nim-lang/nimsuggest.git", "dist/nimsuggest"
```
Comment (MikeMcQuaid) [inline Formula/nim.rb]: Please use `resource` blocks for these so they are cached.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/nim.rb [hunk 3]
```diff
@@ -16,24 +16,28 @@ class Nim < Formula
...
       system "/bin/sh", "build.sh"
     end
     system "/bin/sh", "install.sh", prefix
 
-    system "bin/nim e install_tools.nims"
+    bin.install_symlink prefix/"nim/bin/nim"
+    bin.install_symlink prefix/"nim/bin/nim" => "nimrod"
 
+    # build and install the tools
+    system "bin/nim", "e", "install_tools.nims"
```
Comment (MikeMcQuaid) [inline Formula/nim.rb]: Is there a longer option that can be used instead of `e`? Can remove the comment, it's self-explanatory from the code.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Comment by ysalmi** [comment]
OK, I'll try this. What's unclear to me from the docs is how do I specify
which directory the download gets unpacked to.

**Comment by ysalmi** [comment]
Sadly not. I will remove the comment.

File: Formula/nim.rb (hunk 2 above)
Comment (ysalmi) [inline Formula/nim.rb]: Done

File: Formula/nim.rb (hunk 3 above)
Comment (ysalmi) [inline Formula/nim.rb]: Unfortunately not. And I have removed the comment now.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Comment by ysalmi** [comment]
Mike, thank you for your guidance on this PR. And keep up the great work on
this amazing project.

**Pull request merged**